### ISSUE TITLE
Creation of Validation Set

### ISSUE THREAD

**Issue by apekshapriya** [ifzhang/FairMOT#218]
Hi @ifzhang 
Thanks for the great work!

I have a question about the split of the train and validation set. Can you help me with this?

Suppose I have two datasets, each having a certain number of sequences of images. 
The way I am splitting the train and validation set is:

Taking first 80% of images from each dataset in  train set.
Taking next 20% of images from each dataset in val set.

The ID of objects remain same in train and val set, ie they are not reinitialized in val set again, therefore an object which has ID 1 in trainset, if it is seen in next 20% of images, its id will be same as 1.

I doubt that this approach is a correct way of split as I am facing the below issue:
 My training loss is decreasing fine, every loss is making sense, but for the validation set, every loss is decreasing, except the ID_loss. There is definitely some issue in validation set as the model training is happening well but val ID loss is not decreasing. 

Thanks

**Comment by apekshapriya** [comment]
Hey @ifzhang 
I tried two experiments:

1st exp:
One was to add loss parameters to the model as you had suggested in another issue:
 optimizer.add_param_group({'params': self.loss.parameters()})
I am keeping the validation ids same as train ids:
Doing all these still results in an increase of validation ID and validation hm loss.

2nd exp:

Trying metric as described in test_emb.py. I am running it at every epoch of training. It runs fine with a sample dataset but fails with the actual dataset due to the heavy allocation of memory to GPU. This line is creating problem.

pdist = torch.mm(embedding, embedding.t()).cpu().numpy()

I have tried optimizing the memory by taking rows and columns of embedding and embedding.t() in batches and then concatenating them but it still needs more memory to do this

Any suggestions on this??

Is there any test metric that can be used easily without taking huge memory or so and can be run online after every epoch of training?

**Comment by ifzhang** [comment]
For the first exp, I'm still not sure what is wrong. 
For the second exp, you can try to change embedding to torch.float16 or use less data to run test_emb, like every 5 frames.
